mergin/cli.py

```diff
@@ -51,132 +68,183 @@ def pretty_diff(diff):
+    else:
+        click.secho(f"auth_token created (use --show_token option to see the token).")
+    return session["token"]
+
+
+def get_client(url=None, auth_token=None, username=None, password=None, show_token=False):
+    """Return Mergin client."""
+    if auth_token is not None:
+        mc = MerginClient(url, auth_token=f"Bearer {auth_token}")
+        # Check if the token has expired or is just about to expire
+        delta = mc._auth_session["expire"] - datetime.now(timezone.utc)
+        if delta.total_seconds() > 5:
+            if show_token:
+                click.secho(f'export MERGIN_AUTH="{auth_token}"')
+            return mc
+    if username and password:
+        auth_token = get_token(url, username, password, show_token=show_token)
+        mc = MerginClient(url, auth_token=f"Bearer {auth_token}")
+    else:
+        click.secho(
+            "Missing authorization data.\n"
+            "Either set environment variables (MERGIN_USERNAME and MERGIN_PASSWORD) "
+            "or specify --username / --password options.\n"
+            "Note: if --username is specified but password is missing you will be prompted for password.",
+            fg="red",
+        )
         return None
-
     return mc
 
 
 def _print_unhandled_exception():
     """ Outputs details of an unhandled exception that is being handled right now """
-    click.secho("Unhandled exception!", fg='red')
+    click.secho("Unhandled exception!", fg="red")
     for line in traceback.format_exception(*sys.exc_info()):
         click.echo(line)
 
 
 @click.group()
-def cli():
-    pass
-
-
-@cli.command()
-@click.argument('url', required=False)
-@click.option('--login', prompt=True)
-@click.option('--password', prompt=True, hide_input=True)
-def login(url, login, password):
-    """Fetch new authentication token. If no URL is specified, the public Mergin instance will be used."""
-    c = MerginClient(url)
-    click.echo("Mergin URL: " + c.url)
-    if not c.is_server_compatible():
-        click.secho(str('This client version is incompatible with server, try to upgrade'), fg='red')
-        return
-    try:
-        session = c.login(login, password)
-    except LoginError as e:
-        click.secho('Unable to log in: ' + str(e), fg='red')
-        return
-
-    print('export MERGIN_URL="%s"' % c.url)
-    print('export MERGIN_AUTH="%s"' % session['token'])
+@click.option(
+    "--url",
+    envvar="MERGIN_URL",
+    default="https://public.cloudmergin.com",
+    help="Mergin server URL. Default is: https://public.cloudmergin.com",
+)
+@click.option("--auth_token", envvar="MERGIN_AUTH", help="Mergin authentication token string")
+@click.option("--username", envvar="MERGIN_USERNAME")
+@click.option("--password", cls=OptionPasswordIfUser, prompt=True, hide_input=True, envvar="MERGIN_PASSWORD")
+@click.option(
+    "--show_token",
+    is_flag=True,
+    help="Flag for echoing the authentication token. Useful for setting the MERGIN_AUTH environment variable.",
+)
```

Comment:
This `--show_token` option does not seem particularly useful... I would remove it.

And have we lost `mergin login` command completely?